### ISSUE TITLE
Switch Time Based Training and Validation to # of Data Moments

### ISSUE THREAD

**Issue by raunakdoesdev** [bddmodelcar/training#10]
To have validation and training done at specific times run validation and training for certain numbers of data moments rather than a time based approach. This will be important for publishing research as this gives consistent results, regardless of how slow the training code runs at a ceesrtain point of time.

Also the entire validation set should be evaluated at the end of each epoch of training. If this means we need to reduce the size of the validation set that is fine, but this feature gives replicatable results. We can very easily compare the training over time of two models over a long period of time.

**Comment by tpankaj** [comment]
Resolved by PR #15
